jaeger-core/src/main/java/io/jaegertracing/internal/JaegerTracer.java

```diff
@@ -315,7 +321,7 @@ private JaegerSpanContext createNewContext(String debugId) {
         }
       }
 
-      return new JaegerSpanContext(id, id, 0, flags);
+      return JaegerTracer.this.tracingFactory.createSpanContext(id, id, 0, flags, Collections.<String, String>emptyMap(), debugId);
```

Comment:
I am seeing a mix of `factory` and `tracingFactory`. If you agree with my above comment to call it `TracingObjectFactory`, then I'd suggest field name `objectFactory`, to be used consistently

Reply:
I thought I renamed all `factory` references to `tracingFactory`. Will change to `objectFactory` throughout.